micrometer-core/src/main/java/io/micrometer/core/instrument/Tag.java

```diff
@@ -15,20 +15,47 @@
  */
 package io.micrometer.core.instrument;
 
+import java.util.function.Supplier;
+
 /**
  * Key/value pair representing a dimension of a meter used to classify and drill into measurements.
  *
  * @author Jon Schneider
+ * @author Simon Scholz
  */
 public interface Tag extends Comparable<Tag> {
     String getKey();
 
     String getValue();
 
+    /**
+     * Create a {@link Tag} instance with static values.
+     *
+     * @param key of the tag
+     * @param value of the tag
+     * @return an instance of {@link Tag}
+     */
     static Tag of(String key, String value) {
         return new ImmutableTag(key, value);
     }
 
+    /**
+     * Create a {@link Tag} instance with dynamic values, which may change during runtime.
+     * <p>
+     * NOTE: If the given valueSupplier returns null, an empty {@link String} will be returned as tag value.
```

Comment:
I would add the comment about the null-behavior also to DynamicValueTag and Tags.of(...).

Reply:
Thanks, I've also added this information to `DynamicValueTag` and `Tags.of(...)`